### ISSUE TITLE
free(): invalid pointer when running tensorflow.python.ops.gen_ragged_array_ops.ragged_cross

### ISSUE THREAD

**Issue by dmc1778** [tensorflow/tensorflow#59368]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Have you reproduced the bug with TF nightly?

No

### Source

binary

### Tensorflow Version

2.10.0

### Custom Code

Yes

### OS Platform and Distribution

Ubuntu 22.04

### Python version

3.9

### Current Behaviour?

```shell
Probably due to empty input arguments.
```

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
import os
import numpy as np
from tensorflow.python.ops import gen_ragged_array_ops
try:
  ragged_values_0_tensor = tf.convert_to_tensor(np.ones([3], dtype=str))
  ragged_values_0 = tf.identity(ragged_values_0_tensor)
  ragged_values = [ragged_values_0,]
  ragged_row_splits_0_tensor = tf.random.uniform([4], minval=-256, maxval=257, dtype=tf.int64)
  ragged_row_splits_0 = tf.identity(ragged_row_splits_0_tensor)
  ragged_row_splits = [ragged_row_splits_0,]
  sparse_indices = []
  sparse_values = []
  sparse_shape = []
  dense_inputs = []
  input_order = "R"
  hashed_output = False
  num_buckets = 0
  hash_key = 956888297470
  out_values_type = 7
  out_row_splits_type = 9
  out = gen_ragged_array_ops.ragged_cross(ragged_values=ragged_values,ragged_row_splits=ragged_row_splits,sparse_indices=sparse_indices,sparse_values=sparse_values,sparse_shape=sparse_shape,dense_inputs=dense_inputs,input_order=input_order,hashed_output=hashed_output,num_buckets=num_buckets,hash_key=hash_key,out_values_type=out_values_type,out_row_splits_type=out_row_splits_type,)
except Exception as e:
  print("Error:"+str(e))
```
```


### Relevant log output

```shell
2023-01-21 13:53:18.602619: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-01-21 13:53:18.736809: E tensorflow/stream_executor/cuda/cuda_blas.cc:2981] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2023-01-21 13:53:19.358808: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer.so.7'; dlerror: libnvinfer.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: :/lib/
2023-01-21 13:53:19.359009: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer_plugin.so.7'; dlerror: libnvinfer_plugin.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: :/lib/
2023-01-21 13:53:19.359018: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Cannot dlopen some TensorRT libraries. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.
2023-01-21 13:53:20.000091: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:980] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-21 13:53:20.036637: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudnn.so.8'; dlerror: libcudnn.so.8: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: :/lib/
2023-01-21 13:53:20.036654: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1934] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
Skipping registering GPU devices...
2023-01-21 13:53:20.036907: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
free(): invalid pointer
Aborted

```
```
</details>

**Comment by nimashiri** [comment]
@tilakrayal I am running fuzz testing, so malicious inputs are machine generated. Would you please run my provided test case and confirm whether you encounter crash?

**Comment by tilakrayal** [comment]
@nimashiri,
When I was trying to run `tensorflow.python.ops.gen_ragged_array_ops.ragged_cross` with the **hash_key = 956888297470** it was failing with the error and colab was also crashed. But when I tried to execute on tf-nightly with the same hash_key, it was executed without any issue/error/colab crash. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/ed561e78fd8567df57d9daec9674accf/untitled922.ipynb).

![image](https://user-images.githubusercontent.com/81610181/216596984-7b2baa0c-921e-4af1-b07d-64017fe67e0a.png)

Thank you!

**Comment by mihaimaruseac** [comment]
This is a vulnerability that needs fixing.

@tilakrayal Given that this is a vulnerability, don't be too eager to close it (by adding the label).

@dmc1778 please stop posting vulns on GitHub. Please consult SECURITY.md for how to properly and ethically disclose them.

**Comment by SuryanarayanaY** [comment]
I have tested the user code on Ubuntu22 VM with Tf2.10v and got the below Error log.Similar behaviour observed with TF2.12v also.
```
F tensorflow/core/framework/tensor_shape.cc:186] Non-OK-status: InitDims(dim_sizes) status: INVALID_ARGUMENT: Expected shape dimensions to be non-negative, got -37
Aborted (core dumped)
```
**With TF2.10V:**
```
(tf2.10) suryanarayanay@surya-ubuntu-22-04:~$ python 59368.py
2023-02-16 12:08:30.981380: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 AVX512F AVX512_VNNI FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-02-16 12:08:31.106816: I tensorflow/core/util/util.cc:169] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2023-02-16 12:08:31.144451: E tensorflow/stream_executor/cuda/cuda_blas.cc:2981] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2023-02-16 12:08:31.839684: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer.so.7'; dlerror: libnvinfer.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: :/home/suryanarayanay/miniconda3/envs/tf2.11/lib/:/home/suryanarayanay/miniconda3/envs/tf2.10/lib/
2023-02-16 12:08:31.839785: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer_plugin.so.7'; dlerror: libnvinfer_plugin.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: :/home/suryanarayanay/miniconda3/envs/tf2.11/lib/:/home/suryanarayanay/miniconda3/envs/tf2.10/lib/
2023-02-16 12:08:31.839800: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Cannot dlopen some TensorRT libraries. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.
2.10.0
2023-02-16 12:08:36.688692: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 AVX512F AVX512_VNNI FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-02-16 12:08:37.995527: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1616] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 38239 MB memory:  -> device: 0, name: NVIDIA A100-SXM4-40GB, pci bus id: 0000:00:04.0, compute capability: 8.0
2023-02-16 12:08:37.997262: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1616] Created device /job:localhost/replica:0/task:0/device:GPU:1 with 38239 MB memory:  -> device: 1, name: NVIDIA A100-SXM4-40GB, pci bus id: 0000:00:05.0, compute capability: 8.0
2023-02-16 12:08:38.724652: F tensorflow/core/framework/tensor_shape.cc:186] Non-OK-status: InitDims(dim_sizes) status: INVALID_ARGUMENT: Expected shape dimensions to be non-negative, got -37
Aborted (core dumped)
(tf2.10) suryanarayanay@surya-ubuntu-22-04:~$ 
```

**With TF2.12v:**
```
(tf) suryanarayanay@surya-ubuntu-22-04:~$ python 59368.py
2023-02-16 12:07:04.341671: I tensorflow/core/util/port.cc:110] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2023-02-16 12:07:04.395996: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX512F AVX512_VNNI FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-02-16 12:07:05.219284: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
2.12.0-rc0
2023-02-16 12:07:12.103832: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1635] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 1310 MB memory:  -> device: 0, name: NVIDIA A100-SXM4-40GB, pci bus id: 0000:00:04.0, compute capability: 8.0
2023-02-16 12:07:12.105499: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1635] Created device /job:localhost/replica:0/task:0/device:GPU:1 with 37609 MB memory:  -> device: 1, name: NVIDIA A100-SXM4-40GB, pci bus id: 0000:00:05.0, compute capability: 8.0
2023-02-16 12:07:12.172012: F tensorflow/core/framework/tensor_shape.cc:201] Non-OK-status: InitDims(dim_sizes) status: INVALID_ARGUMENT: Expected shape dimensions to be non-negative, got -143
Aborted (core dumped)
(tf) suryanarayanay@surya-ubuntu-22-04:~$ 
```
Thanks!

**Comment by sachinprasadhs** [comment]
Hi, The issue has been fixed by the commit here https://github.com/tensorflow/tensorflow/commit/bbbe095a3ae243d10752ca16b914a8fa11221d07.
Now it is throwing the proper error message, please find the attached gist [here](https://colab.sandbox.google.com/gist/sachinprasadhs/59d91bef30cac1933aa1950b1f007977/59368.ipynb) for reference.
